botocore/config.py

```diff
@@ -102,6 +102,17 @@ class Config(object):
           this value to 0 will result in no retries ever being attempted on
           the initial request. If not provided, the number of retries will
           default to whatever is modeled, which is typically four retries.
+
+    :type client_cert: str, (str, str)
```

Comment:
I'm definitely open to switching this to be two separate kwargs if we think that's a better interface.

Reply:
I like the string with an optional tuple for the following reasons:

1) The client certificate will essentially be required if a client key is provided. Optional keyword arguments are not great at expressing this. A tuple where the key is the last element better implies this.

2) It follows the precedent used by `requests`. While sure we no longer rely on `requests`, it is still a popular library and by following suit with what they do will make users more comfortable in learning how to use our interface. We currently do this with the `verify` client arg.

I think there were more related options involved it would almost make sense to expose a new ssl config object that you can pass in.